src/store/sagas/wallet.ts

```diff
@@ -116,28 +118,47 @@ export function* handleAirdrop(): Generator {
     })
   )
 }
+
+export function* setEmptyAccounts(
+  collateralsAddresses: PublicKey[]
+): Generator {
+  const tokensAccounts = yield* select(accounts)
+  const acc: PublicKey[] = []
+  for (const collateral of collateralsAddresses) {
+    const collateralTokenProgram = yield* call(getToken, collateral)
+    const accountAddress = tokensAccounts[collateral.toString()]
+      ? tokensAccounts[collateral.toString()].address
+      : null
+    if (accountAddress == null) {
+      acc.push(collateralTokenProgram.publicKey)
+    }
+  }
+  if (acc.length !== 0) {
+    yield* call(createAccount, acc)
+  } else {
+  }
+}
+
 export function* getCollateralTokenAirdrop(
-  collateralTokenAddress: PublicKey,
-  quantity: number
+  collateralsAddresses: PublicKey[],
+  collateralsQuantities: number[]
 ): Generator {
   const wallet = yield* call(getWallet)
-  const tokensAccounts = yield* select(accounts)
-  const collateralTokenProgram = yield* call(getToken, collateralTokenAddress)
-  let accountAddress = tokensAccounts[collateralTokenAddress.toString()]
-    ? tokensAccounts[collateralTokenAddress.toString()].address
-    : null
-  if (accountAddress == null) {
-    accountAddress = yield* call(createAccount, collateralTokenProgram.publicKey)
+  const instructions: Transaction[] = []
+  for (const [index, collateral] of collateralsAddresses.entries()) {
+    yield* call(setEmptyAccounts, collateralsAddresses)
```

Comment:
why you call this every time in the loop?

Reply:
done